packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -705,42 +721,64 @@ export class GarbageCollector implements IGarbageCollector {
 
         // Log all the GC options and the state determined by the garbage collector. This is interesting only for the
         // summarizer client since it is the only one that runs GC. It also helps keep the telemetry less noisy.
-        const gcConfigProps = JSON.stringify({
-            gcEnabled: this.gcEnabled,
-            sweepEnabled: this.sweepEnabled,
-            runGC: this.shouldRunGC,
-            runSweep: this.shouldRunSweep,
-            writeAtRoot: this._writeDataAtRoot,
-            testMode: this.testMode,
-            sessionExpiry: this.sessionExpiryTimeoutMs,
-            inactiveTimeout: this.inactiveTimeoutMs,
-            existing: createParams.existing,
-            trackGCState: this.trackGCState,
-            ...this.gcOptions,
-        });
         if (this.isSummarizerClient) {
             this.mc.logger.sendTelemetryEvent({
                 eventName: "GarbageCollectorLoaded",
-                gcConfigs: gcConfigProps,
+                gcConfigs: JSON.stringify({
+                    gcEnabled: this.gcEnabled,
+                    sweepEnabled: this.sweepEnabled,
+                    runGC: this.shouldRunGC,
+                    runSweep: this.shouldRunSweep,
+                    writeAtRoot: this._writeDataAtRoot,
+                    testMode: this.testMode,
+                    sessionExpiry: this.sessionExpiryTimeoutMs,
+                    inactiveTimeout: this.inactiveTimeoutMs,
+                    existing: createParams.existing,
+                    trackGCState: this.trackGCState,
+                    ...this.gcOptions,
+                }),
             });
         }
+    }
 
-        // Initialize the base state that is used to detect when inactive objects are used.
-        if (this.shouldRunGC) {
```

Comment:
Put this back as `if (this.shouldRunGC && this.isSummarizerClient)`?

Reply:
Also consider putting the `shouldRunGC` check inside the promise's callback so you callers don't have to think about it,